Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -180,22 +182,38 @@ function isNullOrEmpty(str) {
 }
 
 // Creates the string that comprises the build summary text, given the location of the /sonar build folder.
-function createSonarQubeBuildSummary(sqBuildFolder: string): string {
-    // Task report is not created for PR builds
+function createSonarQubeBuildSummary(sqBuildFolder:string):Q.Promise<string> {
+    // Task report is not created for PR builds - inform the user in the build summary
     if (isPrBuild()) {
+        tl.debug('Pull request build, will not attempt to fetch detailed SonarQube report');
         // Looks like: Detailed SonarQube reports are not available for pull request builds.
-        return tl.loc('sqAnalysis_BuildSummaryNotAvailableInPrBuild');
+        return Q.when(tl.loc('sqAnalysis_BuildSummaryNotAvailableInPrBuild'));
     }
 
-    var taskReport: TaskReport = getSonarQubeTaskReport(sqBuildFolder);
-    if (!taskReport) {
-        throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
-    }
+    // Wait for analysis to finish so that we can get analysis information from the SonarQube server
+    var timeout:number = 300;
+    var delay:number = 1;
 
-    var linkToDashBoard: string = createLinkToSonarQubeDashboard(sqBuildFolder);
+    var sqRunSettings:SonarQubeRunSettings = getSonarQubeTaskReport(sqBuildFolder);
```

Comment:
report-task.txt is build agnostic. Its location is build specifc. Even so, we should have little or no code in that common.ts file. We'll refactor as we go along. This seems like an easy win.